x/market/client/cli/tx.go

```diff
@@ -52,17 +54,21 @@ $ terracli market swap --offer-coin="1000krw" --ask-denom="usd"
 			}
 
 			fromAddress := cliCtx.GetFromAddress()
-			fromAccount, err := cliCtx.GetAccount(fromAddress)
-			if err != nil {
-				return err
-			}
 
-			if fromAccount.GetCoins().AmountOf(offerCoin.Denom).LT(offerCoin.Amount) {
-				return fmt.Errorf(strings.TrimSpace(`
-					account %s has insufficient amount of coins to pay the offered coins.\n
-					Required: %s\n
-					Given:    %s\n`),
-					fromAddress, offerCoin, fromAccount.GetCoins())
+			offline := viper.GetBool(flagOffline)
```

Comment:
What purpose does the offline flag serve here? 

Should be added to the long description above if not removing

Reply:
offline flag can be used to building and signing tx without internet connection. Moreover it is useful to test msg building without fully checking account existence or validations. In our test code, we use offline flag to check validity of tx functions.